Java/core/src/main/java/com/amazon/randomcutforest/store/PointStoreFloat.java

```diff
@@ -40,8 +41,19 @@ public PointStoreFloat(boolean overlapping, int startOfFreeSegment, int dimensio
     }
 
     public PointStoreFloat(int dimensions, int capacity) {
-        super(dimensions, 1, capacity, false, true);
-        store = new float[capacity * dimensions];
+        this(new Builder().dimensions(dimensions).shingleSize(1).capacity(capacity).currentCapacity(capacity));
+    }
+
+    @Override
+    void resizeStore() {
+        int maxCapacity = rotationEnabled ? 2 * capacity : capacity;
```

Comment:
Same question: why have this method silently do nothing when it's already at max capacity?

Reply:
This is just a mirror of the double[] case. We are doing something -- if rotation is enabled, this implementation allows itself to go to twice the stated capacity.